HEN_HOUSE/egs++/ausgab_objects/egs_radiative_splitting/egs_radiative_splitting.h

```diff
@@ -106,21 +106,77 @@ class EGS_RADIATIVE_SPLITTING_EXPORT EGS_RadiativeSplitting : public EGS_AusgabO
     ~EGS_RadiativeSplitting();
 
     void setApplication(EGS_Application *App);
-
+    /*! Switch for splitting + RR. Negative nsplit value switches OFF RR. */
     void setSplitting(const int &n_s) {
         nsplit = n_s;
+       if ( nsplit < 0 ){
+          nsplit *= -1;
+          i_play_RR = false;
+       }
+       else if ( nsplit > 1 ){
+          i_play_RR = true;
+       }
+       /* Avoid zero division. A zero value turns off brems */
+       wthin = nsplit ? 1./nsplit : 1.0;
     };
 
-    int processEvent(EGS_Application::AusgabCall iarg) {
-        return 0;
+    bool needsCall(EGS_Application::AusgabCall iarg) const {
+        if (
+             iarg == EGS_Application::BeforeBrems       ||
+             iarg == EGS_Application::BeforeAnnihFlight ||
+             iarg == EGS_Application::BeforeAnnihRest   ||
+             iarg == EGS_Application::AfterBrems        ||
+             iarg == EGS_Application::AfterAnnihFlight  ||
+             iarg == EGS_Application::AfterAnnihRest    ||
+             iarg == EGS_Application::FluorescentEvent  ){
+            return true;
+        }
+        else {
+            return false;
+        }
     };
-    int processEvent(EGS_Application::AusgabCall iarg, int ir) {
+
+    int processEvent(EGS_Application::AusgabCall iarg) {
+
+        /* A fat particle's weight is larger than a thin particle's max weight */
+        bool is_phat = (app->top_p.wt - wthin) > epsilon;
+        bool is_primary = app->top_p.latch == 0 ? true : false;
```

Comment:
Is this compatible with a phase-space from BEAMnrc where the latch is already set? E.g. it's a primary particle but the latch is non-zero due to one of the bits being set by passage.

Reply:
@rtownson I will let you "resolve this conversation if you trust that it is ok...

Reply:
Yes, we can resolve this. Currently egs++ actually zeros out the latch when it's read from a phase-space.